pkg/reconciler/v1alpha1/revision/resolve.go

```diff
@@ -59,11 +60,27 @@ func newResolverTransport(path string) (*http.Transport, error) {
 		return nil, errors.New("Failed to append k8s cert bundle to cert pool.")
 	}
 
-	return &http.Transport{
-		TLSClientConfig: &tls.Config{
-			RootCAs: pool,
-		},
-	}, nil
+	// Copied from https://github.com/golang/go/blob/release-branch.go1.12/src/net/http/transport.go#L42-L53
+	// We want to use the DefaultTransport but change its TLSClientConfig. There
+	// isn't a clean way to do this yet: https://github.com/golang/go/issues/26013
+	resolverTransport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	resolverTransport.TLSClientConfig = &tls.Config{
```

Comment:
move this to the initialization from the getgo?

Reply:
Also, perhaps just:
```
return http.Transport{
			TLSClientConfig: {...},
			Proxy:           http.ProxyFromEnvironment,
		}
```
I think the others can be omitted and will automatically get defaults.

Reply:
Related is #1238 -- I think even what I have here isn't enough, but omitting timeouts means it will wait forever.

cc @dprotaso what's the right setup for a transport here?

Reply:
It seems from that issue the crucial bit is to set the `ResponseHeaderTimeout`. That doesn't have a default I guess because that'd impose a more broader "timeout until the server starts answering" which is very application specific.

Seems like you really have to copy & own the default transport though if you need to change something.

Reply:
> move this to the initialization from the getgo?

You mean move this to `init()`?

Reply:
I think he meant to move the TLSClientConfig into the initialization of `resolverTransport` as shown in the code sample. You could then remove the intermediary variable and simply `return &http.Transport...`